blockstore_caboose.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"crypto/tls"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/filecoin-saturn/caboose"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
+)
+
+func newCabooseBlockStore(orchestrator, loggingEndpoint string, cdns *cachedDNS) (blockstore.Blockstore, error) {
+	var (
+		orchURL *url.URL
+		loggURL *url.URL
+		err     error
+	)
+
+	if orchestrator != "" {
+		orchURL, err = url.Parse(orchestrator)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if loggingEndpoint != "" {
+		loggURL, err = url.Parse(loggingEndpoint)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	saturnServiceClient := &http.Client{
+		Timeout: caboose.DefaultSaturnRequestTimeout,
+		Transport: &withUserAgent{
+			RoundTripper: &http.Transport{
+				DialContext: cdns.dialWithCachedDNS,
+			},
+		},
+	}
+
+	saturnRetrievalClient := &http.Client{
+		Timeout: caboose.DefaultSaturnRequestTimeout,
+		Transport: &withUserAgent{
+			RoundTripper: &http.Transport{
+				// Increasing concurrency defaults from http.DefaultTransport
+				MaxIdleConns:        1000,
+				MaxConnsPerHost:     100,
+				MaxIdleConnsPerHost: 100,
+				IdleConnTimeout:     90 * time.Second,
+
+				DialContext: cdns.dialWithCachedDNS,
+
+				// Saturn Weltschmerz
+				TLSClientConfig: &tls.Config{
+					// Saturn use TLS in controversial ways, which sooner or
+					// later will force them to switch away to different domain
+					// name and certs, in which case they will break us. Since
+					// we are fetching raw blocks and dont really care about
+					// TLS cert being legitimate, let's disable verification
+					// to save CPU and to avoid catastrophic failure when
+					// Saturn L1s suddenly switch to certs with different DNS name.
```

Comment:
I'm not sure if this is correct:
1. It doesn't save any significant CPU. All crypto operations are still the same. You're only saving the check if the chain ends in the root CA store, and some trivial string matching on the domain name.
2. Saturn switching to a different cert should be fine, as long as they're doing a redirect. If the plan is to use a cert for domain X on domain Y, well, that's just plain wrong.

Reply:
Ack on 1, skipping these checks is just a bonus.  
As for 2,  given last week's incidents, and internal conversations, bifrost-gateway does not take any unnecessary risk with Saturn's TLS/DNS setup.